packages/core/src/tracing/tracingUtil.js

```diff
@@ -276,3 +276,26 @@ exports.findCallback = (/** @type {string | any[]} */ originalArgs) => {
     callbackIndex
   };
 };
+
+/**
+ * @param {import('../core').InstanaBaseSpan} span - The span to update
+ * @param {Error} error
+ * @param {string} technology - The technology name (e.g., 'mysql', 'pg', 'http')
+ */
+// @ts-ignore
+exports.setErrorStack = function setErrorStack(span, error, technology) {
+  if (error == null) {
+    return undefined;
+  }
+
+  if (technology && span.data[technology]) {
+    // for some cases like http, it is already set with custom values and no need to overwrite the message
+    span.data[technology].error = span.data[technology].error || exports.getErrorDetails(error);
```

Comment:
IMO a refactoring PR is needed before this PR to make that behavior equal everywhere:

from

> data = { error:

to

> span.data.x.error = ...

Then you can remove this extra check here. Its too confusing.